diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt

```diff
@@ -296,6 +308,69 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
                 }
     }
 
+    /**
+     * Checks that members of [VALUE_PARAMETER_LIST] (list of function parameters at declaration site) are separated with newlines.
+     * Also checks that entries of [SUPER_TYPE_LIST] are separated by newlines.
+     */
+    private fun handleList(node: ASTNode) {
+        if (node.elementType == VALUE_PARAMETER_LIST && node.treeParent.elementType.let { it == FUNCTION_TYPE || it == FUNCTION_TYPE_RECEIVER }) {
+            // do not check other value lists
+            return
+        }
+
+        val (numEntries, entryType) = when (node.elementType) {
+            VALUE_PARAMETER_LIST -> (node.psi as KtParameterList).parameters.size to "value parameters"
+            SUPER_TYPE_LIST -> (node.psi as KtSuperTypeList).entries.size to "supertype list entries"
+            else -> {
+                log.warn("Unexpected node element type ${node.elementType}")
```

Comment:
I guess we have decided that such warnings can be raised only in critical cases. This not that one :)

Reply:
It's not critical, but rather highly improbable. I just wanted to make sure that if this branch ever executes, we have data on why we got there.